src/fileformats/ocd_file_export.cpp

```diff
@@ -2453,45 +2453,38 @@ void OcdFileExport::exportTemplates(OcdFile<Format>& /*file*/)
 
 void OcdFileExport::exportTemplates()
 {
+	static auto const supported_extensions = std::array<const char* const, 8>{{
+	    "bmp", "gif", "jpg", "jpeg", "ocd", "pdf", "tif", "tiff"
+	}};
+	
 	for (int i = map->getNumTemplates() - 1; i >= 0; --i)
 	{
 		const auto* temp = map->getTemplate(i);
-		QString template_path = temp->getTemplatePath();
 		
-		auto supported_by_ocd = false;
-		if (qstrcmp(temp->getTemplateType(), "TemplateImage") == 0)
+		auto const suffix = QFileInfo(temp->getTemplatePath()).suffix().toLower().toUtf8();
+		if (std::find(begin(supported_extensions), end(supported_extensions), suffix) == end(supported_extensions))
+		{
+			addWarning(::OpenOrienteering::OcdFileExport::tr("Unable to export template: file type of \"%1\" is not supported yet")
+			           .arg(temp->getTemplateFilename()));
+			continue;
+		}
+		
+		if (temp->isTemplateGeoreferenced() && suffix != "ocd")
 		{
-			supported_by_ocd = true;
+			if (temp->getTemplateState() == Template::Unloaded)
+			{
+				// Try to load the template, so that the positioning gets set.
+				const_cast<Template*>(temp)->loadTemplateFile(false);
```

Comment:
It may take more memory than available, possibly leading to a crash. This is mitigated by the fact that maps loaded from OCD files will not have georeferenced templates ATM.

We eventually can improve that for some templates but not for all
- by loading only the required georeferencing data from the template.
- by caching georeferencing data even after unloading a template.
- by saving templates as georeferenced to OCD.

Reply:
FTR, this is also what is already implemented before this change, for `TemplateImage`.

Reply:
I agree that it was the previous behavior. I've had a closer look at the load times and memory consumption and while the code behavior is openly wasteful, I haven't noticed any complanints. Let's leave it as is for the moment.